GDJS/Runtime/runtimegame.ts

```diff
@@ -715,24 +729,37 @@ namespace gdjs {
           return;
         }
 
+        const now = Date.now();
+        notYetSentDuration += now - lastSessionResumeTime;
+        lastSessionResumeTime = now;
+
         // Group repeated calls to sendSessionHit - which could
         // happen because of multiple event listeners being fired.
-        if (Date.now() - lastSessionHitTime < 3 * 1000) {
+        if (notYetSentDuration < 5 * 1000) {
           return;
         }
-        lastSessionHitTime = Date.now();
+        const onlySeconds = Math.floor(notYetSentDuration / 1000) * 1000;
+        sendedDuration += onlySeconds;
+        notYetSentDuration -= onlySeconds;
+
         navigator.sendBeacon(
           baseUrl + '/session-hit',
           JSON.stringify({
             gameId: this._data.properties.projectUuid,
             playerId: this._playerId,
             sessionId: this._sessionId,
+            duration: Math.floor(sendedDuration / 1000),
           })
         );
       };
       if (typeof navigator !== 'undefined' && typeof document !== 'undefined') {
         document.addEventListener('visibilitychange', () => {
-          sendSessionHit();
+          if (document.visibilityState === 'visible') {
```

Comment:
if we don't send the hit when the state is not visible, should the logic be the opposite?

Reply:
We send the hit when state becomes hidden and we don't send it when it gets visible back.